target/scripts/helpers/postfix.sh

```diff
@@ -91,3 +94,27 @@ function _vhost_ldap_support() {
 #
 # /etc/aliases is handled by `alias.sh` and uses `postalias` to update the Postfix alias database. No need for `postmap`.
 # http://www.postfix.org/postalias.1.html
+
+# Add an key with an value to Postfix's main configuration file
+# or update an existing key. An already existing key can be updated
+# by either appending to the existing value (default) or by prepending.
+#
+# @param ${1} = key name in Postfix's main configuration file
+# @param ${2} = new value (appended or prepended)
+# @param ${3} = "append" (default) or "prepend" [OPTIONAL]
+function _add_to_or_update_postfix_main() {
+  local KEY=${1:?Key name is required}
+  local NEW_VALUE=${2:?New value is required}
+  if grep -q -E "^${KEY}" "${POSTFIX_MAIN_CF}"; then
+    KEY=$(_escape_for_sed "${KEY}")
+    NEW_VALUE=$(_escape_for_sed "${NEW_VALUE}")
+    if [[ ${3:-append} == 'append' ]]; then
+      local SED_STRING="/${NEW_VALUE}/! s|^(${KEY} *=.*)|\1 ${NEW_VALUE}|g"
+    else
+      local SED_STRING="/${NEW_VALUE}/! s|^(${KEY}) *= *(.*)|\1 = ${NEW_VALUE} \2|g"
+    fi
+    sed -i -E "${SED_STRING}" "${POSTFIX_MAIN_CF}"
+  else
+    echo "${KEY} = ${NEW_VALUE}" >>"${POSTFIX_MAIN_CF}"
+  fi
+}
```

Comment:
FWIW, there is similar functionality already implemented (_although more thoroughly_) in the DB helpers:

https://github.com/docker-mailserver/docker-mailserver/blob/5bada0a83b0875b36934e1de856b724979c83346/target/scripts/helpers/database/db.sh#L33-L37

https://github.com/docker-mailserver/docker-mailserver/blob/5bada0a83b0875b36934e1de856b724979c83346/target/scripts/helpers/database/db.sh#L63-L76

---

It could potentially support `main.cf`, but the overlap introduced here may be better as `db.sh` is already quite complicated and makes some assumptions like:

https://github.com/docker-mailserver/docker-mailserver/blob/5bada0a83b0875b36934e1de856b724979c83346/target/scripts/helpers/database/db.sh#L44-L54

https://github.com/docker-mailserver/docker-mailserver/blob/5bada0a83b0875b36934e1de856b724979c83346/target/scripts/helpers/database/db.sh#L104-L111

with special handling for managing keys with list values (_notably removal, which is less necessary elsewhere_):

https://github.com/docker-mailserver/docker-mailserver/blob/5bada0a83b0875b36934e1de856b724979c83346/target/scripts/helpers/database/db.sh#L126-L135

https://github.com/docker-mailserver/docker-mailserver/blob/5bada0a83b0875b36934e1de856b724979c83346/target/scripts/helpers/database/db.sh#L78-L96

Reply:
I agree that working with `db.sh` would introduce overhead (which I can currently not afford). I ahve further simplified `_add_to_or_update_postfix_main` after seeing your suggestion. I also added some breathing room to make everything more readable.